it/common/src/main/java/org/apache/beam/it/common/PipelineOperator.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.it.common;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkState;
+
+import com.google.auto.value.AutoValue;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+import org.apache.beam.it.common.PipelineLauncher.JobState;
+import org.apache.beam.sdk.function.ThrowingConsumer;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utilities for managing Dataflow jobs. */
+public final class PipelineOperator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PipelineOperator.class);
+
+  /** The result of running an operation. */
+  public enum Result {
+    CONDITION_MET,
+    LAUNCH_FINISHED,
+    LAUNCH_FAILED,
+    TIMEOUT
+  }
+
+  private final PipelineLauncher client;
+
+  public PipelineOperator(PipelineLauncher client) {
+    this.client = client;
+  }
+
+  /**
+   * Waits until the given job is done, timing out it if runs for too long.
+   *
+   * <p>If the job is a batch job, it should complete eventually. If it is a streaming job, this
+   * will time out unless the job is explicitly cancelled or drained.
+   *
+   * @param config the configuration for performing the operation
+   * @return the result, which will be {@link Result#LAUNCH_FINISHED}, {@link Result#LAUNCH_FAILED}
+   *     or {@link Result#TIMEOUT}
+   */
+  @SuppressWarnings("rawtypes")
```

Comment:
Prefer to suppress this on a single line that creates the `Supplier[]` and assigns it to a variable with a proper type like `Supplier<Boolean>[]`